src/js/common/util.js

```diff
@@ -0,0 +1,161 @@
+fluid.registerNamespace("gpii.devpmt");
+
+/**
+ * binderCheckboxToBoolean
+ *
+ * By default GPII binder, when bound to a checkbox input will
+ * represent checked as an array of one item, and unchecked as
+ * an empty array. For many situations we would like these to
+ * be represented as true and false. These transforms accomplish
+ * that mapping.
+ *
+ * @param (Array) value - The array coming from binder, most likely []
+ * or ['on']
+ * @return (boolean) true or false based on the array size
+ */
+gpii.devpmt.binderCheckboxToBoolean = function (value) {
+    return value.length > 0;
+};
+
+fluid.defaults("gpii.devpmt.binderCheckboxToBoolean", {
+    gradeNames: ["fluid.standardTransformFunction"]
+});
+
+/**
+ * booleanToBinderCheckbox
+ *
+ * See `gpii.devpmt.binderCheckboxToBoolean` for a description. This
+ * is the reverse transform.
+ *
+ * @param (boolean) value - Current value as a boolean
+ * @return (Array) [] or ['on'] depending on value
+ */
+gpii.devpmt.booleanToBinderCheckbox = function (value) {
+    return value ? ["on"] : [];
+};
+
+fluid.defaults("gpii.devpmt.booleanToBinderCheckbox", {
+    gradeNames: ["fluid.standardTransformFunction"]
+});
+
+gpii.devpmt.booleanBinderRules = {
+    domToModel: {
+        "": {
+            transform: {
+                type: "gpii.devpmt.binderCheckboxToBoolean",
+                inputPath: ""
+            }
+        }
+    },
+    modelToDom: {
+        "": {
+            transform: {
+                type: "gpii.devpmt.booleanToBinderCheckbox",
+                inputPath: ""
+            }
+        }
+    }
+};
+
+/**
+ * contextNames - Takes a rawPrefs set and returns a list of all the
+ * context names. This function ensures that the first item in the list is
+ * always the default context, 'gpii-default', and the rest of the contexts
+ * in an order that will always be the same. The stability of the order is
+ * so that we can build html tables and other structures successfully.
+ *
+ * @param prefs (Object) - A prefs set.
+ * @return (Array) - List of context names.
+ */
+gpii.devpmt.contextNames = function (prefs) {
+    var contextNames = [];
+    fluid.each(prefs.contexts, function (value, key) {
+        if (key !== "gpii-default" && contextNames.indexOf(key) < 0) {
+            contextNames.push(key);
+        }
+    });
+    contextNames.sort();
+    contextNames.unshift("gpii-default");
+    return contextNames;
+};
+
+
+/**
+ * npsetApplications - Goes through all the contexts in a flat prefs set
+ * and pulls out the flat pref URI and app ID. Returns all the apps a
+ * user has settings.
+ *
+ * @param prefs (Object) Transformed flat preferences set
+ * @return (Array) An array of objects structured as the following: (RWG example)
+ *     {
+ *         uri: "http://registry.gpii.net/applications/com.texthelp.readWriteGold",
+ *         appId: "com.texthelp.readWriteGold"
+ *     }
+ */
+gpii.devpmt.npsetApplications = function (prefs) {
```

Comment:
Likewise, globally eliminate the use of "NP set"

Reply:
Occurrences of this curse have also been removed.